packages/internal-test-utils/consoleMock.js

```diff
@@ -418,13 +418,18 @@ export function createLogAssertion(
         let argIndex = 0;
         // console.* could have been called with a non-string e.g. `console.error(new Error())`
         // eslint-disable-next-line react-internal/safe-string-coercion
-        String(format).replace(/%s/g, () => argIndex++);
+        String(format).replace(/%s|%c/g, () => argIndex++);
         if (argIndex !== args.length) {
-          logsMismatchingFormat.push({
-            format,
-            args,
-            expectedArgCount: argIndex,
-          });
+          if (format.includes('%c%s')) {
```

Comment:
Can this mirror the check in `shouldIgnoreConsoleError` i.e. 
```suggestion
          if (format.startsWith('%c%s%c')) {
```
or are these files unrelated?

To be honest, I don't quite follow why we need to ignore this case and why there's no good default for our test suite. Could you add test to `packages/internal-test-utils/__tests__/ReactInternalTestUtils-test.js` to illustrate what this is doing?

Reply:
In `shouldIgnoreConsoleError` the more specific one was ok because we didn't test any paths that used the server printer:

https://github.com/sebmarkbage/react/blob/ba2389424a1deb617e62d8473e4c2e718a2ed984/packages/react-client/src/ReactClientConsoleConfigServer.js#L13

But tbh, none of this makes any sense. We should find a way to remove `shouldIgnoreConsoleError`. There must be some systemic issue in our tests that we can address so we don't need this anymore. I keep just hacking in whatever I need to keep it running.

Regarding the argument mismatch I feel like we should just remove this assertion. We've found a case where this is valid. So if we keep it we need some way to disable it. Isn't the lint warning enough for the common mistakes and we can disable the lint warning as needed?

Reply:
We can use the usual `assertConsoleErrorDev` for now but probably want a specific `assertConsoleBadgedErrorDev` in the future to encode the behavior clearer: https://github.com/sebmarkbage/react/pull/6/